### DISCUSSION TITLE
How to scan for basic int value

### DISCUSSION THREAD

**Discussion by ascorbic-acid** [fuqiuluo/Hakutaku#15] [Q&A] [answered]
Hello, thank you for providing this nice tool. Unfortunately, I am having trouble getting the basic integer search to work on my device. I'm not sure if I am doing something wrong, but here is the code I am using:

```
using namespace hak;

int main()
{
    auto pid = hak::find_process(packageName);
    auto process = std::make_shared<hak::process>(pid);
    process->set_memory_mode(memory_mode::SYSCALL);
    auto searcher = hak::memory_searcher(process);


    auto size = searcher.searchNumber("555", type_i32, EQ);
    std::cout << "search result size: " << size << "\n";
 }
```

error:
`terminating with uncaught exception of type hak::file_not_found: File not found.
`

**Comment by fuqiuluo** [comment, marked as answer]
```c++
std::ifstream maps(std::string("/proc/") + std::to_string(pid) + "/maps");
if (!maps.is_open()) {
    throw hak::file_not_found();
}
```
Before starting the search, Hak needs to obtain the content of the process maps. If the object cannot be found (the process does not exist) or the permissions are insufficient, it will prompt that it cannot be opened.

**Comment by ascorbic-acid** [comment]
thanks, it was my bad i forgot to use sudo inside termux
